### PR TITLE
Update artifact finder to support round5 with auto move gear spaceship

### PR DESCRIPTION
**Pull request by snowtigersoft**
Auto move Gear to the nearest Foundry for prospect and find

### CODE REVIEW

File: content/artifacts/artifacts-finder/plugin.js [hunk 1]
```diff
@@ -200,3 +230,9 @@ class ArtifactsFinder {
 }
 
 export default ArtifactsFinder;
+
+function distance(from, to) {
+  let fromloc = from.location;
+  let toloc = to.location;
+  return Math.sqrt((fromloc.coords.x - toloc.coords.x) ** 2 + (fromloc.coords.y - toloc.coords.y) ** 2);
+}
```
Comment (cristobal) [inline content/artifacts/artifacts-finder/plugin.js:238]: Perhaps this code block code be move before the ArtifactsFinder class with the rest of the utility functions at the top of the file, to maintain the readability in the file.

File: content/artifacts/artifacts-finder/plugin.js [hunk 2]
```diff
@@ -120,36 +136,50 @@ class ArtifactsFinder {
...
       return;
     }
-    let currentBlockNumber = df.contractsAPI.ethConnection.blockNumber;
-    let planets = Array.from(df.getMyPlanets()).filter(canHaveArtifact);
-    planets.forEach(planet => {
-      if (isFindable(planet, currentBlockNumber)) {
-        this.pendingPlanets.push(planet);
+    if (isProspectable(gearPlanet)) {
+      if (!gearPlanet.transactions?.hasTransaction(isUnconfirmedProspectPlanetTx)) {
+        df.prospectPlanet(gearPlanet.locationId);
```
Comment (cristobal) [inline content/artifacts/artifacts-finder/plugin.js:148]: I hade done something similar in another plugin with await/async syntax:
```js
    const uctx = await df.prospectPlanet(planet.locationId)
    await uctx.confirmedPromise // wait until confirmed
    while (df.getPlanetWithId(planet.locationId).prospectedBlockNumber === undefined) {
      await sleep(blockTimeInMs)
    }
```
The code above guarantees will wait until the planet has finished prospecting.

**Review by ichub — DISMISSED** [review]

**Comment by ichub** [comment]
One final thing before we merge this - could you update the compatibility version of this plugin to `0.6.5`?

**Comment by snowtigersoft** [comment]
@ichub Updated the  compatibility version of this plugin to 0.6.5

**Review by ichub — APPROVED** [review]

**Pull request merged**